class Solution {
    public int[] findMissingAndRepeatedValues(int[][] grid) {
        int n=grid.length;
        int totalElement=n*n;

        int[] freq = new int[totalElement+1];
        for(int i=0; i<n; i++){
            for(int j=0; j<n; j++){
                freq[grid[i][j]]++;
            }
        }

        int repeat=-1;
        int missing=-1;
        for(int i=1; i<=totalElement; i++){
            if(freq[i]==2){
                repeat=i;
            }
            else if(freq[i]==0){
                missing=i;
            }
        }

        return new int[]{repeat, missing};
    }
}